Zed crashes just by pressing "g" while editing a C file

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Use zed 0.163.0 (f5cbfa718e5bb4a284d0c2bfac6b425e4d7a343d) (but the bug may be older), macOS build.

Open a C file.

Press the g key (no modifiers, just "g").

Zed dies.

(Edit: it does not happen with 0.161.2)

### Environment

Zed: v0.163.0 (Zed Dev f5cbfa718e5bb4a284d0c2bfac6b425e4d7a343d)
OS: macOS 15.2.0
Memory: 24 GiB
Architecture: aarch64


### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-11-19T18:42:45.169675+01:00 [INFO] ========== starting zed ==========
2024-11-19T18:42:45.204794+01:00 [INFO] Opening main db
2024-11-19T18:42:45.206088+01:00 [INFO] Opening main db
2024-11-19T18:42:45.208518+01:00 [INFO] Using git binary path: Some("/Applications/Zed Dev.app/Contents/MacOS/git")
2024-11-19T18:42:45.30066+01:00 [INFO] set environment variables from shell:/bin/zsh, path:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin
2024-11-19T18:42:45.389713+01:00 [ERROR] theme not found: Melange Modified Light
2024-11-19T18:42:45.390305+01:00 [INFO] No prompt template overrides directory found at /Users/mocenigo/.config/zed/prompt_overrides. Using built-in prompts.
2024-11-19T18:42:45.390715+01:00 [INFO] extensions updated. loading 15, reloading 0, unloading 0
2024-11-19T18:42:45.420271+01:00 [INFO] Opening main db
2024-11-19T18:42:45.428175+01:00 [ERROR] theme not found: Melange Modified Light
2024-11-19T18:42:45.430468+01:00 [ERROR] theme not found: Melange Modified Light
2024-11-19T18:42:45.43074+01:00 [INFO] set status on client 0: Authenticating
2024-11-19T18:42:45.433127+01:00 [ERROR] theme not found: Melange Modified Light
2024-11-19T18:42:45.43376+01:00 [WARN] Theme "macOS Classic Light" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-11-19T18:42:45.433816+01:00 [WARN] Theme "macOS Classic Dark" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-11-19T18:42:45.433865+01:00 [WARN] Theme "macOS Classic Dark2" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-11-19T18:42:45.459604+01:00 [INFO] set status on client 137487: Connecting
2024-11-19T18:42:45.46047+01:00 [INFO] Opening main db
2024-11-19T18:42:45.465003+01:00 [INFO] open paths ["/Users/mocenigo/.config/zed/settings.json"]
2024-11-19T18:42:45.465488+01:00 [INFO] Opening main db
2024-11-19T18:42:45.496861+01:00 [INFO] attempting to start language server "yaml-language-server", path: "/Users/mocenigo/Research/PAC-Analysis/code/differentialModelSAT/examples/qarmav1_r_333_t_1_z_4_zc_6_zcb_(8,10,12,13,15)_W32_bsw_2.yaml", id: 1
2024-11-19T18:42:45.497525+01:00 [INFO] fetching latest version of language server "yaml-language-server"
2024-11-19T18:42:45.505403+01:00 [INFO] attempting to start language server "json-language-server", path: "/Users/mocenigo/.config/zed/settings.json", id: 2
2024-11-19T18:42:45.507194+01:00 [INFO] fetching latest version of language server "json-language-server"
2024-11-19T18:42:45.555631+01:00 [INFO] attempting to start language server "clangd", path: "/Users/mocenigo/Research/PAC-Analysis/code_unused/poisson_differential.c", id: 3
2024-11-19T18:42:45.689162+01:00 [INFO] using project environment variables shell launched in "/Users/mocenigo/m.sh". PATH="/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/opt/homebrew/sbin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/opt/homebrew/bin:/Users/mocenigo/bin:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin"
2024-11-19T18:42:45.689985+01:00 [INFO] found user-installed language server for clangd. path: "/usr/bin/clangd", arguments: []
2024-11-19T18:42:45.690064+01:00 [INFO] starting language server process. binary path: "/usr/bin/clangd", working directory: "/Users/mocenigo/Research/PAC-Analysis/code_unused", args: []
2024-11-19T18:42:45.807855+01:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-11-19T18:42:46.534624+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.535821+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.536999+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.538108+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.53831+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.538343+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.538376+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.538408+01:00 [WARN] no language server found matching 'texlab'
2024-11-19T18:42:46.53877+01:00 [INFO] add_connection;
2024-11-19T18:42:46.544267+01:00 [INFO] set status on client 137487: Connected { peer_id: PeerId { owner_id: 611, id: 633258 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-11-19T18:42:47.156484+01:00 [INFO] attempting to start language server "scls", path: "/Users/mocenigo/Research/PAC-Analysis/updates.md", id: 4
2024-11-19T18:42:47.156541+01:00 [INFO] attempting to start language server "typos", path: "/Users/mocenigo/Research/PAC-Analysis/updates.md", id: 5
2024-11-19T18:42:47.156586+01:00 [INFO] attempting to start language server "typos", path: "/Users/mocenigo/.config/zed/settings.json", id: 6
2024-11-19T18:42:47.156628+01:00 [INFO] attempting to start language server "scls", path: "/Users/mocenigo/Research/PAC-Analysis/code/differentialModelSAT/examples/qarmav1_r_333_t_1_z_4_zc_6_zcb_(8,10,12,13,15)_W32_bsw_2.yaml", id: 7
2024-11-19T18:42:47.156669+01:00 [INFO] attempting to start language server "typos", path: "/Users/mocenigo/Research/PAC-Analysis/code/differentialModelSAT/examples/qarmav1_r_333_t_1_z_4_zc_6_zcb_(8,10,12,13,15)_W32_bsw_2.yaml", id: 8
2024-11-19T18:42:47.156718+01:00 [INFO] attempting to start language server "texlab", path: "/Users/mocenigo/Research/PAC-Analysis/paper_short", id: 9
2024-11-19T18:42:47.156766+01:00 [INFO] attempting to start language server "scls", path: "/Users/mocenigo/Research/PAC-Analysis/code_unused/poisson_differential.c", id: 10
2024-11-19T18:42:47.156806+01:00 [INFO] attempting to start language server "typos", path: "/Users/mocenigo/Research/PAC-Analysis/code_unused/poisson_differential.c", id: 11
2024-11-19T18:42:47.156853+01:00 [INFO] attempting to start language server "scls", path: "/Users/mocenigo/m.sh", id: 12
2024-11-19T18:42:47.1569+01:00 [INFO] attempting to start language server "typos", path: "/Users/mocenigo/m.sh", id: 13
2024-11-19T18:42:47.15769+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/.cargo/bin/simple-completion-language-server", working directory: "/Users/mocenigo/Research/PAC-Analysis", args: []
2024-11-19T18:42:47.158067+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/.cargo/bin/simple-completion-language-server", working directory: "/Users/mocenigo/Research/PAC-Analysis/code/differentialModelSAT/examples", args: []
2024-11-19T18:42:47.158284+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/.cargo/bin/simple-completion-language-server", working directory: "/Users/mocenigo/Research/PAC-Analysis/code_unused", args: []
2024-11-19T18:42:47.158466+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/.cargo/bin/simple-completion-language-server", working directory: "/Users/mocenigo", args: []
2024-11-19T18:42:47.15862+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/texlab", working directory: "/Users/mocenigo/Research/PAC-Analysis/paper_short", args: []
2024-11-19T18:42:47.176162+01:00 [WARN] unhandled capability registration: Registration { id: "pull-config", method: "workspace/didChangeConfiguration", register_options: None }
2024-11-19T18:42:47.192424+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/", args: ["/Users/mocenigo/Library/Application Support/Zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2024-11-19T18:42:47.406403+01:00 [INFO] downloading language server "yaml-language-server"
2024-11-19T18:42:47.596204+01:00 [INFO] downloading language server "json-language-server"
2024-11-19T18:42:47.651221+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/Users/mocenigo/Research/PAC-Analysis/code/differentialModelSAT/examples", args: ["/Users/mocenigo/Library/Application Support/Zed/languages/yaml-language-server/node_modules/yaml-language-server/bin/yaml-language-server", "--stdio"]
2024-11-19T18:42:47.684395+01:00 [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.236.0 initialized"
}
2024-11-19T18:42:47.684474+01:00 [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
{
  "items": [
    {
      "section": "github.copilot"
    },
    {
      "section": "github-enterprise"
    },
    {
      "section": "http"
    },
    {
      "section": "telemetry"
    }
  ]
}
2024-11-19T18:42:47.763346+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/Library/Application Support/Zed/extensions/work/typos/typos-lsp-v0.1.30/typos-lsp", working directory: "/Users/mocenigo/Research/PAC-Analysis", args: []
2024-11-19T18:42:47.764791+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/Library/Application Support/Zed/extensions/work/typos/typos-lsp-v0.1.30/typos-lsp", working directory: "/Users/mocenigo/.config/zed", args: []
2024-11-19T18:42:47.765051+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/Library/Application Support/Zed/extensions/work/typos/typos-lsp-v0.1.30/typos-lsp", working directory: "/Users/mocenigo/Research/PAC-Analysis/code/differentialModelSAT/examples", args: []
2024-11-19T18:42:47.765508+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/Library/Application Support/Zed/extensions/work/typos/typos-lsp-v0.1.30/typos-lsp", working directory: "/Users/mocenigo/Research/PAC-Analysis/code_unused", args: []
2024-11-19T18:42:47.765752+01:00 [INFO] starting language server process. binary path: "/Users/mocenigo/Library/Application Support/Zed/extensions/work/typos/typos-lsp-v0.1.30/typos-lsp", working directory: "/Users/mocenigo", args: []
2024-11-19T18:42:47.850042+01:00 [WARN] unhandled capability registration: Registration { id: "29f71866-af6d-44f6-8868-dbfcd13db6e0", method: "workspace/didChangeWorkspaceFolders", register_options: Some(Object {}) }
2024-11-19T18:42:47.850132+01:00 [WARN] unhandled capability registration: Registration { id: "66c5464b-4026-48c0-a212-26e4331878dc", method: "workspace/didChangeConfiguration", register_options: Some(Object {}) }
2024-11-19T18:42:47.883574+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/Users/mocenigo/.config/zed", args: ["/Users/mocenigo/Library/Application Support/Zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2024-11-19T18:42:57.788256+01:00 [INFO] ========== starting zed ==========
2024-11-19T18:42:57.81659+01:00 [INFO] Opening main db
2024-11-19T18:42:57.817205+01:00 [INFO] Opening main db
2024-11-19T18:42:57.818043+01:00 [INFO] Using git binary path: Some("/Applications/Zed Dev.app/Contents/MacOS/git")
2024-11-19T18:42:57.921316+01:00 [INFO] set environment variables from shell:/bin/zsh, path:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/opt/homebrew/sbin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin:/opt/homebrew/bin:/Users/mocenigo/bin:/Users/mocenigo/Research-PyEnv/bin:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin://Applications/Topaz Photo AI Beta.app/Contents/Resources/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin
2024-11-19T18:42:57.995485+01:00 [ERROR] theme not found: Melange Modified Light
2024-11-19T18:42:57.996064+01:00 [INFO] No prompt template overrides directory found at /Users/mocenigo/.config/zed/prompt_overrides. Using built-in prompts.
2024-11-19T18:42:57.996303+01:00 [INFO] extensions updated. loading 15, reloading 0, unloading 0
2024-11-19T18:42:58.021926+01:00 [INFO] Opening main db
2024-11-19T18:42:58.028236+01:00 [ERROR] theme not found: Melange Modified Light
2024-11-19T18:42:58.03133+01:00 [INFO] set status on client 0: Authenticating
2024-11-19T18:42:58.03296+01:00 [WARN] Theme "macOS Classic Light" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-11-19T18:42:58.033018+01:00 [WARN] Theme "macOS Classic Dark" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-11-19T18:42:58.033069+01:00 [WARN] Theme "macOS Classic Dark2" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-11-19T18:42:58.061869+01:00 [INFO] set status on client 137487: Connecting
2024-11-19T18:42:58.062734+01:00 [INFO] Opening main db
2024-11-19T18:42:58.095232+01:00 [INFO] using project environment variables from CLI. PATH="/Users/mocenigo/Research-PyEnv/bin:/Users/mocenigo/.zi/polaris/sbin:/Users/mocenigo/.zi/polaris/bin:/opt/homebrew/opt/cython/bin:/opt/homebrew/opt/bison/bin:/Users/mocenigo/.cargo/bin:/Library/Java/JavaVirtualMachines/graalvm-23.jdk/Contents/Home/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/Users/mocenigo/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin://Applications/Topaz Photo AI Beta.app/Contents/Resources/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/mocenigo/Library/Python/3.8/bin:/opt/homebrew/opt/llvm/bin:/Users/mocenigo/.local/bin:/Users/mocenigo/.ghc/bin:/Users/mocenigo/.cabal/bin:/Users/mocenigo/.ghcup/bin:/opt/homebrew/opt/icu4c/sbin:/opt/homebrew/opt/icu4c/bin"
2024-11-19T18:42:58.101812+01:00 [INFO] Opening main db
2024-11-19T18:42:58.105817+01:00 [INFO] open paths ["/Users/mocenigo/.config/zed/settings.json"]
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/editor/src/editor.rs
+++ b/crates/editor/src/editor.rs
@@ -1036,7 +1036,12 @@ impl CompletionsMenu {
         let match_candidates = completions
             .iter()
             .enumerate()
-            .map(|(id, completion)| StringMatchCandidate::new(id, completion.label.text.clone()))
+            .map(|(id, completion)| {
+                StringMatchCandidate::new(
+                    id,
+                    completion.label.text[completion.label.filter_range.clone()].into(),
+                )
+            })
             .collect();
 
         Self {

